component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java

```diff
@@ -306,24 +320,21 @@ private Event convertToSingleEventForDefaultMapping(Object eventObject) throws I
         while (!parser.isClosed()) {
             JsonToken jsonToken = parser.nextToken();
             if (JsonToken.START_OBJECT.equals(jsonToken)) {
-                parser.nextToken();
+                jsonToken = parser.nextToken();
                 if (DEFAULT_JSON_EVENT_IDENTIFIER.equalsIgnoreCase(parser.getText())) {
-                    parser.nextToken();
-                } else {
-                    log.error("Default json message " + eventObject
-                            + " contains an invalid event identifier. Required \"event\", " +
-                            "but found \"" + parser.getText() + "\". Hence dropping the message.");
-                    return null;
+                    jsonToken = parser.nextToken();
                 }
-            } else if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+            }
+            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
```

Comment:
What happen's if the event is like 
{\"testEvent\":{\"symbol\":\"WSO2\",\"price\":56.6,\"volume\":99}} 

We should not allow it to be mapped via default mapping

Reply:
If fail.on.missing.attribute option is true, that event will be dropped.
If fail.on.missing.attribute option is false, the null values will be assigned to the event attributes.

Should we drop such events in the first place without considering fail.on.missing.attribute option?